lib/std/json.zig

```diff
@@ -1573,7 +1573,7 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
                                     //     .UseLast => {},
                                     // }
                                     if (options.duplicate_field_behavior == .UseFirst) {
-                                        break;
+                                        // do nothing, check of fields_seen[i] below will parse value without overwriting field
```

Comment:
don't you also need a check in the `field.is_comptime` branch?

Reply:
I'm assuming that for comptime fields, the parsed value must always equal the field's `default_value`, even when the key is a duplicate that will be thrown out. For example, if parsing to a `struct { comptime x: u32 = 99 }` with `UseFirst` then parsing `{ "x": 99, "x": 99 }` is fine but `{ "x": 99, "x": 100 }` should return `UnexpectedValue` even though the second value is ostensibly being ignored. Then a comptime field should have the same behavior for the first field and for duplicate fields when either `UseFirst` or `UseLast` is set.

Writing this, I have noticed a nasty problem with `parseFree`: if a struct has a comptime field that is or contains a pointer of any kind, then `parseFree` will try to deallocate that memory even though the comptime field holds only its default value and not anything that was allocated. And this problem also extends to the `UseLast` case I recently adjusted.

Reply:
> I'm assuming that for comptime fields, the parsed value must always equal the field's `default_value`, even when the key is a duplicate that will be thrown out. For example, if parsing to a `struct { comptime x: u32 = 99 }` with `UseFirst` then parsing `{ "x": 99, "x": 99 }` is fine but `{ "x": 99, "x": 100 }` should return `UnexpectedValue` even though the second value is ostensibly being ignored. Then a comptime field should have the same behavior for the first field and for duplicate fields when either `UseFirst` or `UseLast` is set.

That behaviour has the downside that changing a field from runtime to comptime has externally visible effects; which would be nice to avoid.
I think with `.UseFirst`,  `{ "x": 99, "x": 100 }`  should ignore the `100` unconditionally.

Reply:
That's a good point, and I think a compelling case that that's the correct interpretation of `UseFirst`. I'll change it.